library/ssl_tls.c

```diff
@@ -6618,7 +6589,9 @@ int mbedtls_ssl_read( mbedtls_ssl_context *ssl, unsigned char *buf, size_t len )
                 /* With DTLS, drop the packet (probably from last handshake) */
 #if defined(MBEDTLS_SSL_PROTO_DTLS)
                 if( ssl->conf->transport == MBEDTLS_SSL_TRANSPORT_DATAGRAM )
-                    return( MBEDTLS_ERR_SSL_WANT_READ );
+                {
+                    continue;
+                }
```

Comment:
@hanno-arm since this is DCP , packets can get lost. What will be the behavior in this case? The comment says "drop the packet", but now, we loop, instead of returning `MBEDTLS_ERR_SSL_WANT_READ` to the caller.